app/models/event_stream.rb

```diff
@@ -54,19 +56,19 @@ def self.event_groups
   end
 
   def self.group_and_level(event_type)
-    group, v = event_groups.find { |_k, v| v[:critical].include?(event_type) || v[:detail].include?(event_type) }
-    if group.nil?
-      group, level = :other, :detail
-    else
-      level = v[:detail].include?(event_type) ? :detail : :critical
+    level = :detail # the level is detail as default
+    group = event_groups.find do |_k, value|
+      lvl_found = GROUP_LEVELS.detect { |lvl| value[lvl]&.include?(event_type) }
```

Comment:
Is ```&``` a typo here?
Where are the ```warning``` events defined? Otherwise ```value[:warning]``` returns ```nil```.

Reply:
hey `&` not is a typo, this handle new values and avoid the flow be stopped when occurs `nil.include?`. Also we can define warning events in the [`settings.yml`](https://github.com/ManageIQ/manageiq-providers-lenovo/blob/master/config/settings.yml) for each provider and I will open a PR to add warning events in the provider lenovo after this PR be merged.

Reply:
@lfu

Reply:
@CharlleDaniel Could you please write some test cases to ensure the format of the return value is the same as before?

Reply:
@lfu ok, no problem.

Reply:
@lfu done, the tests were created.